Elph/sync-branch
# feat: upgrade dependencies and add new tests

### Overview

This PR updates our GitHub Action, `Elph/sync-branch@v1.3.0`, to Node.js 20, addressing the deprecation warning from GitHub regarding Node.js 16 actions. The update aims to ensure our action remains compatible with GitHub's current runtime environment and addresses potential issues with the `set-output` command deprecation. We've also upgraded dependencies, renamed a module for consistency, and implemented previously commented tests.

### Changes

#### Node.js Version Upgrade
- Upgraded the runtime environment to Node.js 20 to align with [GitHub's transition requirements](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/).

#### Dependency Upgrades
- Updated `@actions/core`, `@actions/github`, `@vercel/ncc`, `eslint`, and `jest` to their latest versions.

#### Code and Test Improvements
- Renamed module `updatePullrequest` to `updatePullRequest` for naming consistency with `createPullRequest`.
- Implemented previously commented tests in `index.test.js` to enhance coverage and reliability.

#### Attempt to Address `set-output` Deprecation
- Made efforts to comply with GitHub's guidance on [migrating from `set-output` to Environment Files](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/). It's important to note that while we aim for backward compatibility, there might be scenarios where this change could affect existing workflows.

### Motivation

The primary motivation behind these updates is to ensure our GitHub Action remains functional, secure, and up to date with the latest practices recommended by GitHub. By upgrading to Node.js 20, we mitigate the risk of future deprecations affecting our action's usability. Additionally, updating our dependencies and improving our test suite directly contributes to the stability and reliability of the action.

### Testing Instructions

1. Checkout this PR locally.
2. Run `npm install` to update to the latest dependencies.
3. Execute `npm test` to run the updated test suite and ensure all tests pass.
4. For actions developers, test the action in a controlled workflow to ensure it executes as expected without deprecation warnings.
